pkg/backup/backup.go

```diff
@@ -420,3 +435,70 @@ func (*realItemBackupper) backupItem(ctx *backupContext, item map[string]interfa
 
 	return nil
 }
+
+func (ib *realItemBackupper) handlePodExecHooks(ctx *backupContext, namespace, name string, item, metadata map[string]interface{}) error {
+	for _, hookSpec := range ctx.backup.Spec.Hooks.Resources {
+		hookNamespaces := collections.NewIncludesExcludes().Includes(hookSpec.Namespaces...)
+		if !hookNamespaces.ShouldInclude(namespace) {
+			continue
+		}
+
+		hookResources := collections.NewIncludesExcludes().Includes(hookSpec.Resources...)
+		if !hookResources.ShouldInclude("pods") {
+			continue
+		}
+
+		if hookSpec.LabelSelector != nil {
+			labelSelector, err := metav1.LabelSelectorAsSelector(hookSpec.LabelSelector)
+			if err != nil {
+				return err
+			}
+
+			temp, err := collections.GetMap(metadata, "labels")
+			if err != nil {
+				return err
+			}
+			podLabels := make(map[string]string, len(temp))
+			for k, v := range temp {
+				podLabels[k] = v.(string)
+			}
+			if !labelSelector.Matches(labels.Set(podLabels)) {
+				continue
+			}
+		}
+
+		for _, hook := range hookSpec.Hooks {
+			if hook.Exec != nil {
+				containers, err := collections.GetSlice(item, "spec.containers")
+				if err != nil {
+					return err
+				}
+				foundContainer := false
+				for _, c := range containers {
+					c2 := c.(map[string]interface{})
+					if c2["name"].(string) == hook.Exec.Container {
```

Comment:
this conversion may not be needed, if c2["name"] _is_ a `string` then equality works as expected.